keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/TagVersioningStrategy.kt

```diff
@@ -17,24 +17,18 @@
  */
 package com.netflix.spinnaker.keel.api
 
-import com.fasterxml.jackson.annotation.JsonProperty
 import com.netflix.spinnaker.keel.api.SortType.INCREASING
 import com.netflix.spinnaker.keel.api.SortType.SEMVER
 
 /**
  * Different options for versioning for tags
  */
-enum class TagVersionStrategy(val regex: String, val sortType: SortType) {
-  @JsonProperty("increasing-tag")
-  INCREASING_TAG("""(^.*$)""", INCREASING),
-  @JsonProperty("semver-tag")
-  SEMVER_TAG("""(^.*$)""", SEMVER),
-  @JsonProperty("branch-job-commit-by-job")
-  BRANCH_JOB_COMMIT_BY_JOB("""^master-h(\d+).*$""", INCREASING), // popular netflix strategy
-  @JsonProperty("semver-job-commit-by-job")
-  SEMVER_JOB_COMMIT_BY_JOB("""^v.*-h(\d+).*$""", INCREASING), // popular netflix strategy
-  @JsonProperty("semver-job-commit-by-semver")
-  SEMVER_JOB_COMMIT_BY_SEMVER("""^v(.*)-h\d+.*$""", SEMVER); // popular netflix strategy
+enum class TagVersionStrategy(val regex: String, val sortType: SortType, val friendlyName: String) {
+  INCREASING_TAG("""(^.*$)""", INCREASING, "increasing-tag"),
+  SEMVER_TAG("""(^.*$)""", SEMVER, "semver-tag"),
+  BRANCH_JOB_COMMIT_BY_JOB("""^master-h(\d+).*$""", INCREASING, "branch-job-commit-by-job"), // popular netflix strategy
+  SEMVER_JOB_COMMIT_BY_JOB("""^v.*-h(\d+).*$""", INCREASING, "semver-job-commit-by-job"), // popular netflix strategy
+  SEMVER_JOB_COMMIT_BY_SEMVER("""^v(.*)-h\d+.*$""", SEMVER, "semver-job-commit-by-semver"); // popular netflix strategy
```

Comment:
@emjburns Side-comment: as much as I guess they're self-explanatory, these names feel too long and easy to get wrong if you're manually creating a spec. Maybe there's an opportunity to simplify a bit? I'd offer suggestions but I'm unfamiliar with the details -- maybe we could add some examples in the comments?

Reply:
Here are the docs about it, with examples: https://www.spinnaker.io/guides/user/managed-delivery/artifacts/#docker-images. I prefer explicit over short in this case.

Reply:
Agreed. After reading the descriptions again, I understand what the names mean. It's unfortunate that the `-by-<field>` at the end is ambiguous, but I can't think of better alternatives. 👍

Reply:
Yeah, I had a similar reaction to these. They didn't feel super meaningful to me.